app/policies/policy_context.rb

```diff
@@ -0,0 +1,12 @@
+class PolicyContext
+  attr_reader :record
+
+  def initialize(record, user)
+    @record = record
+    @user = user
+  end
+
+  def policy_class
+    "#{@user.roles.uniq.first}DefenceRequestPolicy".classify
```

Comment:
I don't have much experience with Pundit, so I'm going through the code trying to understand it. But if get it correctly, this approach won't allow a user to have 2 different roles on the same controller, will it? Cause the `before_action` filter checks a single policy against the requested action. So if I'm a _solicitor_ and _company admin_, one of the roles does allow access to an action and the other doesn't, it won't work as expected.